nomos-services/data-availability/verifier/src/lib.rs

```diff
@@ -0,0 +1,205 @@
+        if let Some(attestation) = storage_adapter.get_attestation(blob).await? {
+            Ok(attestation)
+        } else {
+            let attestation = verifier.verify(blob)?;
+            storage_adapter.add_blob(blob, &attestation).await?;
+            Ok(attestation)
+        }
+    }
+
+    async fn should_stop_service(message: LifecycleMessage) -> bool {
+        match message {
+            LifecycleMessage::Shutdown(sender) => {
+                if sender.send(()).is_err() {
+                    error!(
+                        "Error sending successful shutdown signal from service {}",
+                        Self::SERVICE_ID
+                    );
+                }
+                true
+            }
+            LifecycleMessage::Kill => true,
+        }
+    }
+}
+
+impl<Backend, N, S> ServiceData for DaVerifierService<Backend, N, S>
+where
+    Backend: VerifierBackend,
+    Backend::Settings: Clone,
+    Backend::Error: Error,
+    N: NetworkAdapter,
+    N::Settings: Clone,
+    S: DaStorageAdapter,
+    S::Settings: Clone,
+{
+    const SERVICE_ID: ServiceId = "DaVerifier";
+    type Settings = DaVerifierServiceSettings<Backend::Settings, N::Settings, S::Settings>;
+    type State = NoState<Self::Settings>;
+    type StateOperator = NoOperator<Self::State>;
+    type Message = DaVerifierMsg<Backend::DaBlob>;
+}
+
+#[async_trait::async_trait]
+impl<Backend, N, S> ServiceCore for DaVerifierService<Backend, N, S>
+where
+    Backend: VerifierBackend + Send + Sync + 'static,
+    Backend::Settings: Clone + Send + Sync + 'static,
+    Backend::DaBlob: Debug + Send + Sync + 'static,
+    Backend::Attestation: Debug + Send,
+    Backend::Error: Error + Send + Sync,
+    N: NetworkAdapter<Blob = Backend::DaBlob, Attestation = Backend::Attestation> + Send + 'static,
+    N::Settings: Clone + Send + Sync + 'static,
+    S: DaStorageAdapter<Blob = Backend::DaBlob, Attestation = Backend::Attestation>
+        + Send
+        + Sync
+        + 'static,
+    S::Settings: Clone + Send + Sync + 'static,
+{
+    fn init(service_state: ServiceStateHandle<Self>) -> Result<Self, DynError> {
+        let DaVerifierServiceSettings {
+            verifier_settings, ..
+        } = service_state.settings_reader.get_updated_settings();
+        let network_relay = service_state.overwatch_handle.relay();
+        let storage_relay = service_state.overwatch_handle.relay();
+        Ok(Self {
+            network_relay,
+            storage_relay,
+            service_state,
+            verifier: Backend::new(verifier_settings),
+        })
+    }
+
+    async fn run(self) -> Result<(), DynError> {
+        // This service will likely have to be modified later on.
+        // Most probably the verifier itself need to be constructed/update for every message with
+        // an updated list of the available nodes list, as it needs his own index coming from the
+        // position of his bls public key landing in the above-mentioned list.
```

Comment:
It need to update itself. This list need to be read from new blocks if it change.

Reply:
(This is your comment) I've also added another one where the verifier is initialized. I'll work on "params provider" service after basic integration between services suceeds.